### ISSUE TITLE
Why different handling between GFF and mzml/genbank in polars.

### ISSUE THREAD

**Issue by tshauck** [wheretrue/biobear#93]
All have maps, but GFF is converted ok into a polars df, but not the latter to.

* https://github.com/wheretrue/exon/blob/b2726a591294ff9b0b45e6632e28727eaa9bc1ce/exon/exon-genbank/src/config.rs#L22

**Comment by abearab** [comment]
@tshauck have you seen https://github.com/BiocPy?

**Comment by tshauck** [comment]
@abearab I've seen it, but not used it. I am a fan of those packages in R (e.g. granges). I also know there's some other analogues in Python for some of those bioconductor packages (e.g. AnnData).

**Comment by abearab** [comment]
> I've seen it, but not used it. I am a fan of those packages in R (e.g. granges). I also know there's some other analogues in Python for some of those bioconductor packages (e.g. AnnData).

Yeah, I do like AnnData and [scverse](https://scverse.org/) ecosystem a lot. However, a good implementation of granges for python has been missing for long time! I thought considering this granges can be very relevant to your implementation of annotation file formats (i.e. GTF, GFF, BED, BAM, etc.). This is just another suggestion, feel free to ignore it :)

**Comment by tshauck** [comment]
100% -- you can actually do a little of granges stuff via SQL joins, but it's not quite as intuitive or as specialized to genomic intervals. E.g. say you ran bakta, and wanted to get CDSs where a spacer annotation is within 100bp of a CDS

```sql
WITH cds AS (
  SELECT *
  FROM gff_scan('bakta.gff')
  WHERE type = 'cds'
), spacers AS (
  SELECT *
  FROM gff_scan('bakta.gff')
  WHERE type = 'crispr-spacer'
)

SELECT *
FROM cds
  JOIN spacers
    ON spacers.start > (cds.start - 100) OR spacers.end < (cds.end - 100)
```
